src/lib/apirequest.ts

```diff
@@ -82,15 +82,34 @@ async function createAPIRequestAsync<T>(parameters: APIRequestParams) {
   );
 
   const media = params.media || {};
-  const resource = params.resource;
+
+  /**
+   * In a previous version of this API, the request body was stuffed in a field
+   * named `resource`.  This caused lots of problems, because it's not uncommon
+   * to have an actual named parameter required which is also named `resource`.
+   * This mean that users would have to use `resource_` in those cases, which
+   * pretty much nobody figures out on their own. The request body is now
+   * documented as being in the `requestBody` property, but we also need to keep
+   * using `resource` for reasons of back-compat. Cases that need to be covered
+   * here:
+   * - user provides just a `resource` with a request body
+   * - user provides both a `resource` and a `resource_`
```

Comment:
The code below does not handle this case at all (probably because it's not really needed to handle it), why mention it here?